agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2413,6 +2432,163 @@ bool backhaul_manager::handle_ap_capability_query(ieee1905_1::CmduMessageRx &cmd
+                    // Fill a query vector
+                    ap_metric_query.push_back({socket->slave, mac});
+                }
+            }
+            i++;
+        }
+    }
+    return ret;
+}
+
+bool backhaul_manager::handle_ap_metrics_response(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                  const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received AP_METRICS_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto ap_metric_tlv = cmdu_rx.getClass<wfa_map::tlvApMetric>();
+    if (!ap_metric_tlv) {
+        LOG(ERROR) << "Failed cmdu_rx.getClass<wfa_map::tlvApMetric>(), mid=" << std::hex
+                   << int(mid);
+        return false;
+    }
+
+    auto bssid_tlv = ap_metric_tlv->bssid();
+    auto mac       = std::find_if(
+        ap_metric_query.begin(), ap_metric_query.end(),
+        [&](sApMetricsQuery const &ap_metric_query) { return ap_metric_query.bssid == bssid_tlv; });
+
+    if (mac == ap_metric_query.end()) {
+        LOG(ERROR) << "Failed search in ap_metric_query for bssid: " << bssid_tlv
+                   << " from mid=" << std::hex << int(mid);
+        return false;
+    }
+
+    sApMetric metric;
+    // Copy data to the response vector
+    metric.bssid               = ap_metric_tlv->bssid();
+    metric.channel_utilization = ap_metric_tlv->channel_utilization();
+    metric.number_of_stas_currently_associated =
+        ap_metric_tlv->number_of_stas_currently_associated();
+    metric.estimated_service_parameters = ap_metric_tlv->estimated_service_parameters();
+    auto info                           = ap_metric_tlv->estimated_service_info_field();
+    if (ap_metric_tlv->estimated_service_info_field_length()) {
+        for (size_t i = 0; i < ap_metric_tlv->estimated_service_info_field_length(); i++) {
+            metric.estimated_service_info_field.push_back(info[i]);
+        }
+    }
+
+    // Fill a response vector
+    ap_metric_response.push_back({mac->soc, mac->bssid, metric});
+
+    // Remove an entry from the processed query
+    ap_metric_query.erase(std::remove_if(ap_metric_query.begin(), ap_metric_query.end(),
+                                         [&](sApMetricsQuery const &ap_metric_query) {
+                                             return mac->bssid == ap_metric_query.bssid;
+                                         }),
+                          ap_metric_query.end());
+
+    processed_queries++;
+    if (!ap_metric_query.empty()) {
+        return true;
+    }
+
+    // If all queries processed - prepare and send response message to the controller
+    auto cmdu_header = cmdu_tx.create(mid, ieee1905_1::eMessageType::AP_METRICS_RESPONSE_MESSAGE);
+
+    if (!cmdu_header) {
+        LOG(ERROR) << "Failed building IEEE1905 AP_METRICS_RESPONSE_MESSAGE";
+        return false;
+    }
+
+    // Prepare tlvApMetric for each processed query
+    for (size_t i = 0; i < processed_queries; i++) {
```

Comment:
We should just iterate over ap_metric_response here.
```suggestion
    for (response : ap_metric_response) {
```